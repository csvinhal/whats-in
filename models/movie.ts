export interface Movie {
    title: string
    imdb: string
    year: string
    poster: string
    favorite: boolean
}

export interface MovieResponse {
    movies?: Movie[]
    totalResults?: string
}

export interface SearchResponse {
    Title: string
    Year: string
    imdbID: string
    Poster: string
}

export interface OmdbMovieResponse {
    Search?: SearchResponse[]
    totalResults?: string
    Error?: string
    Response: 'True' | 'False'
}
