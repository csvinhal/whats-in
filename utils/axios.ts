import axios from 'axios'

const axiosInstance = axios.create({
    baseURL: process.env.BASE_URL,
})

const backendAxiosInstance = axios.create({
    baseURL: process.env.backendUrl,
})

export { axiosInstance, backendAxiosInstance }
