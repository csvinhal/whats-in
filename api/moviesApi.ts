import { MovieResponse } from '../models/movie'
import { MovieDetails } from '../models/movieDetails'
import axios from '../utils/axios'

export const getMovies = (title: string) => {
    return axios.get<MovieResponse>(`/movies`, {
        params: { title },
    })
}

export const getMovieDetail = (slug: string) => {
    return axios.get<MovieDetails>(`/movie-detail/${slug}`)
}

export const updateMovieFavorite = (slug: string, favorite: boolean) => {
    return axios.post<MovieDetails>(`/movie-detail/${slug}`, { favorite })
}
