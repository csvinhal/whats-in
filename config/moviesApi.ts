import { MovieResponse } from '../models/movie'
import { MovieDetails } from '../models/movieDetails'
import axios from '../utils/axios'

export const getMovies = (title: string) => {
    return axios.get<MovieResponse>(`/api/movies`, {
        params: { title },
    })
}

export const getMovieDetail = (slug: string) => {
    return axios.get<MovieDetails>(`/api/movie-detail/${slug}`)
}

export const updateMovieFavorite = (slug: string, favorite: boolean) => {
    return axios.post<MovieDetails>(`/api/movie-detail/${slug}`, { favorite })
}
