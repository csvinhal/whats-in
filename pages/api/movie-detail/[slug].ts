import axios from 'axios'
import { NextApiRequest, NextApiResponse } from 'next'
import Favorite from '../../../models/favorite'
import { DetailResponse, MovieDetails } from '../../../models/movieDetails'
import dbConnect from '../../../utils/mongo'

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
    const { method } = req

    switch (method) {
        case 'GET':
            return handleGet(req, res)
        case 'POST':
            return handlePost(req, res)
    }
}

const handleGet = async (
    req: NextApiRequest,
    res: NextApiResponse,
): Promise<void> => {
    const { slug } = req.query

    const response = await axios.get<DetailResponse>(
        `http://www.omdbapi.com/?i=${slug}&apikey=${process.env.API_KEY}&plot=full`,
    )

    if (hasError(response.data.Response)) {
        return res.status(500).json(response.data)
    }

    const parsedMovie = await parseMovie(response.data)

    return res.status(200).json(parsedMovie)
}

const hasError = (response: 'True' | 'False'): boolean => {
    return response === 'False'
}

const parseMovie = async (movie: DetailResponse): Promise<MovieDetails> => {
    const rottenRation = movie.Ratings.find(rate =>
        rate.Source.toLowerCase().includes('rotten'),
    )
    const favorite = await getFavorite(movie.imdbID)
    return {
        title: movie.Title,
        imdb: movie.imdbID,
        year: movie.Year,
        poster: movie.Poster.includes('http')
            ? movie.Poster
            : `/${movie.Poster}`,
        rated: movie.Rated,
        runtime: movie.Runtime,
        plot: movie.Plot,
        actors: movie.Actors.split(','),
        genre: movie.Genre.split(','),
        director: movie.Director.split(','),
        imdbRate: `${movie.imdbRating}/10`,
        rottenRate: rottenRation ? `${rottenRation.Value}` : null,
        favorite,
    }
}

const getFavorite = async (slug: string): Promise<boolean> => {
    await dbConnect()
    return await Favorite.exists({ slug })
}

const handlePost = async (req: NextApiRequest, res: NextApiResponse) => {
    const { slug } = req.query
    const { favorite } = req.body
    await dbConnect()

    if (favorite) {
        await Favorite.findOneAndRemove({ slug: slug as string })
        return res.status(200).json({ favorite: false })
    }

    await Favorite.create({ slug: slug as string })
    return res.status(200).json({ favorite: true })
}

export default handler
