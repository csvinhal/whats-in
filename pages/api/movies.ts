import axios from 'axios'
import { NextApiRequest, NextApiResponse } from 'next'
import Favorite from '../../models/favorite'
import {
    Movie,
    MovieResponse,
    OmdbMovieResponse,
    SearchResponse,
} from '../../models/movie'
import dbConnect from '../../utils/mongo'

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
    if (req.method === 'GET') {
        try {
            const { title } = req.query
            const response = await axios.get<OmdbMovieResponse>(
                `${process.env.omdbApi}/?s=${title}&type=movie&apikey=${process.env.API_KEY}`,
            )

            if (hasError(response.data.Response)) {
                return handleError(res, response.data.Error)
            }

            const parsedData = await parseResponse(response.data)
            return res.status(200).json(parsedData)
        } catch (e: unknown) {
            return res.status(500).json({
                title: 'Ops! Something went wrong.',
                subtitle: 'Please try again later.',
            })
        }
    }
    return res.status(400)
}

function hasError(response: 'True' | 'False'): boolean {
    return response === 'False'
}

function handleError(res: NextApiResponse, error: string) {
    switch (error) {
        case 'Movie not found!':
            return res.status(404).json({
                title: "What a pity! We don't found this movie.",
                subtitle: 'Try to search for another movie.',
            })
        case 'Too many results.':
            return res.status(500).json({
                title: 'We found too many results.',
                subtitle:
                    'Please try your search again with more specific keywords.',
            })
        default:
            return res.status(500).json({
                title: 'Ops! Something went wrong.',
                subtitle: 'Please try again later.',
            })
    }
}

const parseResponse = async (
    data: OmdbMovieResponse,
): Promise<MovieResponse> => {
    const parsedMovies = await parseMovie(data.Search)

    return {
        movies: parsedMovies,
        totalResults: data.totalResults,
    }
}

const parseMovie = async (omdbMovies: SearchResponse[]): Promise<Movie[]> => {
    const imdbs = omdbMovies.map(movie => movie.imdbID)

    await dbConnect()

    const favorites = await Favorite.find({ slug: { $in: imdbs } })

    return omdbMovies.map(movie => ({
        title: movie.Title,
        imdb: movie.imdbID,
        year: movie.Year,
        poster: movie.Poster.includes('http')
            ? movie.Poster
            : `/${movie.Poster}`,
        favorite: !!favorites.find(favorite => favorite.slug === movie.imdbID),
    }))
}

export default handler
