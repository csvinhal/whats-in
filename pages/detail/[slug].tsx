import axios from 'axios'
import {
    GetServerSideProps,
    GetServerSidePropsContext,
    InferGetServerSidePropsType,
} from 'next'
import Image from 'next/image'
import { useCallback, useMemo, useState } from 'react'
import FavoriteButton from '../../components/FavoriteButton/FavoriteButton'
import Grid from '../../components/Grid/Grid'
import GridItem from '../../components/GridItem/GridItem'
import Labels from '../../components/Labels/Labels'
import Layout from '../../components/Layout/Layout'
import SummaryCredit from '../../components/SummaryCredit/SummaryCredit'
import SummaryLabels from '../../components/SummaryLabels/SummaryLabels'
import Typography from '../../components/Typography/Typography'
import { updateMovieFavorite } from '../../config/moviesApi'
import { MovieDetails } from '../../models/movieDetails'

export const MovieDetail = ({
    movie,
}: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element => {
    const [movieDetail, setMovieDetail] = useState(movie)
    const toggleFavorite = useCallback(async () => {
        const { favorite } = movieDetail
        const { data } = await updateMovieFavorite(movieDetail.imdb, favorite)
        setMovieDetail({ ...movieDetail, favorite: data.favorite })
    }, [movieDetail, setMovieDetail])

    const favoriteLabel = useMemo(
        () => (movieDetail.favorite ? 'Added' : 'Add to favourites'),
        [movieDetail.favorite],
    )

    return (
        <Layout className="mb-3" showBackIcon>
            <Grid
                className="summary"
                flexDirection="column"
                flexDirectionMd="row-reverse"
                justifyContentMd="space-between"
            >
                <GridItem md={6}>
                    <div className="summary__image">
                        <Image
                            src={movieDetail.poster}
                            layout="responsive"
                            width={480}
                            height={640}
                        />
                    </div>
                </GridItem>

                <GridItem md={6}>
                    <div className="summary__content">
                        <SummaryLabels
                            className="mb-3"
                            runtime={movieDetail.runtime}
                            rated={movieDetail.rated}
                            year={movieDetail.year}
                        />

                        <Typography className="summary__title mb-3" size="bold">
                            {movieDetail.title}
                        </Typography>

                        <div className="summary__rates">
                            <Labels className="mb-3" logo="imdb">
                                {movieDetail.imdbRate}
                            </Labels>
                            {movieDetail.rottenRate && (
                                <Labels className="mb-3" logo="rotten">
                                    {movieDetail.rottenRate}
                                </Labels>
                            )}
                            <FavoriteButton
                                className="mb-3"
                                favorite={movieDetail.favorite}
                                onClick={toggleFavorite}
                                aria-label={favoriteLabel}
                            >
                                {favoriteLabel}
                            </FavoriteButton>
                        </div>

                        <div className="mb-3">
                            <Typography className="mb-2" color="secundary">
                                Plot
                            </Typography>
                            <Typography>{movieDetail.plot}</Typography>
                        </div>

                        <Grid
                            flexDirection="column"
                            alignItems="center"
                            flexDirectionMd="row"
                            justifyContentMd="space-around"
                        >
                            <GridItem>
                                <SummaryCredit
                                    className="summary__credit mt-1"
                                    title="Cast"
                                    list={movieDetail.actors}
                                />
                            </GridItem>
                            <GridItem>
                                <SummaryCredit
                                    className="summary__credit mt-1"
                                    title="Genre"
                                    list={movieDetail.genre}
                                />
                            </GridItem>
                            <GridItem>
                                <SummaryCredit
                                    className="summary__credit mt-1"
                                    title="Director"
                                    list={movieDetail.director}
                                />
                            </GridItem>
                        </Grid>
                    </div>
                </GridItem>
            </Grid>

            <style jsx>{`
                    .summary__rates {
                        display: flex;
                        flex-direction: row;
                        flex-wrap: wrap;
                    }
    
                    .summary__rates > :global(*:not(last-children)) {
                        margin-right: 1rem;
                    }
    
                    :global(.summary__credit) {
                        text-align: center;
                    }
    
                    .summary__image {
                        position: relative;
                        overflow: hidden;
                        width: 160px;
                        height: auto;
                        border-radius: 8px;
                        margin: 0 auto 16px auto;
                    }

                    @media (max-width: 767px) {
                        .summary__content :global(.summary__title) {
                            font-size: var(--theme-font-size-medium-high);
                            font-weight: var(--theme-font-weight-medium-high);
                            line-height: var(--theme-line-height-medium-high);
                            letter-spacing: var(--theme-letter-spacing-medium-high);
                        }
                    }
    
                    @media (min-width: 768px) {

                        :global(.back__button) {
                            margin-bottom 24px;
                        }
    
                        .content__detail {
                            padding: 8px;
                        }
    
                        .summary__image {
                            width: 480px;
                            max-height: 640px;
                            border-radius: 8px;
                            margin-left: auto;
                        }
    
                        :global(.summary__credit) {
                            text-align: left;
                        }
                    }
                `}</style>
        </Layout>
    )
}

export const getServerSideProps: GetServerSideProps<{
    movie: MovieDetails
}> = async (context: GetServerSidePropsContext) => {
    const { params } = context
    const res = await axios.get<MovieDetails>(
        `/api/movie-detail/${params.slug}`,
    )

    if (!res.data) {
        return {
            notFound: true,
        }
    }

    return {
        props: { movie: res.data },
    }
}
export default MovieDetail
