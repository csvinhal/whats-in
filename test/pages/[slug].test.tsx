import { fireEvent, waitFor } from '@testing-library/dom'
import { AxiosResponse } from 'axios'
import { toMatchDiffSnapshot } from 'snapshot-diff'
import * as moviesApi from '../../api/moviesApi'
import { MovieDetail } from '../../pages/detail/[slug]'
import { render } from '../testUtils'
import movie from './__mocks__/movie-detail.json'

expect.extend({ toMatchDiffSnapshot })

const axiosResponse: AxiosResponse = {
    data: { favorite: true },
    status: 200,
    statusText: 'OK',
    config: {},
    headers: {},
}

describe('Movie detail page', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<MovieDetail movie={movie} />, {})
        expect(asFragment()).toMatchSnapshot()
    })

    it('matches snapshot', () => {
        const { asFragment, rerender } = render(
            <MovieDetail movie={movie} />,
            {},
        )
        const firstRender = asFragment()
        const favoriteMovie = { ...movie, favorite: true }
        rerender(<MovieDetail movie={favoriteMovie} />)
        expect(firstRender).toMatchDiffSnapshot(asFragment())
    })

    it('should add movie to favorite', async () => {
        jest.spyOn(moviesApi, 'updateMovieFavorite').mockImplementation(() =>
            Promise.resolve(axiosResponse),
        )

        const { getByText } = render(<MovieDetail movie={movie} />, {})

        const favoriteButton = getByText('Add to favourites')
        expect(favoriteButton).toBeTruthy()

        fireEvent.click(favoriteButton)

        await waitFor(() =>
            expect(moviesApi.updateMovieFavorite).toHaveBeenCalled(),
        )

        expect(getByText('Added')).toBeTruthy()
    })
})
