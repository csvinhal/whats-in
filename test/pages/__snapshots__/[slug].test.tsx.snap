// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie detail page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2234572585 layout"
  >
    <header
      class="jsx-2234572585 mb-3"
    >
      <nav
        class="jsx-2234572585"
      >
        <button
          aria-label="Return to home page"
          class="jsx-4102583334 button layout__back"
        >
          <svg
            class="button__icon"
            height="1.5rem"
            version="1.1"
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="Styleguide-presentation"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="8.icons"
                transform="translate(-794.000000, -344.000000)"
              >
                <g
                  id="icon__arrow"
                  transform="translate(794.000000, 344.000000)"
                >
                  <rect
                    height="24"
                    id="Rectangle-13"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <polyline
                    id="Rectangle-3"
                    points="4 17 4 7 4 7 14 7"
                    stroke="currentColor"
                    stroke-width="2"
                    transform="translate(9.000000, 12.000000) rotate(-45.000000) translate(-9.000000, -12.000000) "
                  />
                  <path
                    d="M3,12 L23,12"
                    id="Line"
                    stroke="currentColor"
                    stroke-linecap="square"
                    stroke-width="2"
                  />
                </g>
              </g>
            </g>
          </svg>
        </button>
        <a
          aria-label="What's in"
          class="jsx-2234572585"
          href="/"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTExIiBoZWlnaHQ9IjMyIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </nav>
    </header>
    <main
      class="jsx-2234572585"
    >
      <button
        aria-label="Return to home page"
        class="jsx-4102583334 button layout__back"
      >
        <svg
          class="button__icon"
          height="1.5rem"
          version="1.1"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Styleguide-presentation"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="8.icons"
              transform="translate(-794.000000, -344.000000)"
            >
              <g
                id="icon__arrow"
                transform="translate(794.000000, 344.000000)"
              >
                <rect
                  height="24"
                  id="Rectangle-13"
                  width="24"
                  x="0"
                  y="0"
                />
                <polyline
                  id="Rectangle-3"
                  points="4 17 4 7 4 7 14 7"
                  stroke="currentColor"
                  stroke-width="2"
                  transform="translate(9.000000, 12.000000) rotate(-45.000000) translate(-9.000000, -12.000000) "
                />
                <path
                  d="M3,12 L23,12"
                  id="Line"
                  stroke="currentColor"
                  stroke-linecap="square"
                  stroke-width="2"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <div
        class="jsx-533276974 grid summary flex-direction-md justify-content-md"
      >
        <div
          class="jsx-2249794958 grid-item grid-item-md-6"
        >
          <div
            class="jsx-3039394109 summary__image"
          >
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 133.33333333333331%;"
              />
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </div>
        <div
          class="jsx-2249794958 grid-item grid-item-md-6"
        >
          <div
            class="jsx-3039394109 summary__content"
          >
            <div
              class="jsx-3813441065 movie-label mb-3"
            >
              <span
                class="jsx-570952775 typography"
              >
                99 min
              </span>
              <span
                class="jsx-3813441065 movie-label__dot"
              />
              <span
                class="jsx-570952775 typography"
              >
                2008
              </span>
              <span
                class="jsx-3813441065 movie-label__dot"
              />
              <div
                class="jsx-3066171714 labels"
              >
                <div
                  class="jsx-3066171714 labels__content"
                >
                  <span
                    class="jsx-4266605292 typography labels__typography"
                  >
                    PG-13
                  </span>
                </div>
              </div>
            </div>
            <p
              class="jsx-2478908140 typography summary__title mb-3"
            >
              What Happens in Vegas
            </p>
            <div
              class="jsx-3039394109 summary__rates"
            >
              <div
                class="jsx-3066171714 labels labels__logo mb-3"
              >
                <div
                  class="jsx-1034338665 imdb-icon"
                >
                  <svg
                    height="1rem"
                    version="1.1"
                    viewBox="0 0 35 16"
                    width="2.188rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      id="Styleguide-presentation"
                      stroke="none"
                      stroke-width="1"
                    >
                      <g
                        fill="#0A1014"
                        fill-rule="nonzero"
                        id="7.Labels"
                        transform="translate(-392.000000, -882.000000)"
                      >
                        <g
                          id="icon__imdb"
                          transform="translate(392.000000, 882.000000)"
                        >
                          <g
                            id="Group"
                            transform="translate(17.500000, 8.463502) scale(-1, 1) rotate(-180.000000) translate(-17.500000, -8.463502) translate(0.000000, 0.934090)"
                          >
                            <polygon
                              id="Shape"
                              points="0 7.32673772 0 0.213658131 1.84939148 0.213658131 3.69878296 0.213658131 3.69878296 7.32673772 3.69878296 14.4398173 1.84939148 14.4398173 0 14.4398173"
                            />
                            <path
                              d="M4.73245436,7.32673772 L4.73245436,0.213658131 L6.36460446,0.213658131 L7.99639959,0.213658131 L8.00172414,4.79118616 C8.00704868,7.30512388 8.01805274,9.32513218 8.0290568,9.28155017 C8.04006085,9.23796817 8.33362069,7.23461315 8.6765213,4.82909896 C9.01906694,2.42358478 9.31298174,0.403930796 9.32931034,0.338734948 C9.356643,0.214012457 9.36196755,0.214012457 10.4989351,0.214012457 C11.6359026,0.214012457 11.6412272,0.214012457 11.6628803,0.338734948 C11.6792089,0.403930796 11.853144,1.65824221 12.0490872,3.11877093 C12.2393509,4.58496886 12.5169371,6.64820484 12.658215,7.70692872 L12.9194726,9.63445813 L12.9358012,4.92122353 L12.9468053,0.213658131 L14.5679513,0.224642215 L16.1834178,0.240941176 L16.1834178,7.32673772 L16.1834178,14.4125343 L13.7792089,14.4288332 L11.3693205,14.4398173 L11.3419878,14.3204097 C11.3309838,14.249899 11.1460446,12.827283 10.9341278,11.1548678 C10.4716024,7.51134118 10.5262677,7.87523322 10.487931,7.97834187 C10.4662779,8.02192388 10.2540061,9.47713772 10.0090771,11.2090796 C9.76414807,12.9410215 9.55223124,14.3746215 9.53590264,14.4015502 C9.52489858,14.423164 8.43691684,14.439463 7.12068966,14.439463 L4.73280933,14.439463 L4.73280933,7.32673772 L4.73245436,7.32673772 Z"
                              id="Shape"
                            />
                            <path
                              d="M17.2440669,7.32106851 L17.2440669,0.202674048 L20.3337221,0.229957093 C23.0045132,0.246256055 23.4723631,0.262555017 23.8259128,0.344049827 C24.8702333,0.582865052 25.4413793,1.10407751 25.6699797,2.02177993 C25.7789554,2.47779654 25.8385903,4.10131488 25.8332657,6.97382976 C25.8332657,10.079845 25.7732759,11.8117869 25.6483266,12.2841024 C25.3544118,13.3973924 24.5276876,14.0649412 23.1841278,14.2715128 C22.4876775,14.3799363 21.0518256,14.4398173 19.1371197,14.4398173 L17.2440669,14.4398173 L17.2440669,7.32106851 Z M21.7642495,11.8550145 C22.1071501,11.6813952 22.1014706,11.703009 22.1288032,8.19518893 C22.1504564,4.8829564 22.1014706,3.42242768 21.9601927,3.0748346 C21.8568966,2.84133426 21.6339757,2.71094256 21.2367647,2.67302976 L20.9428499,2.64043183 L20.9428499,7.32638339 L20.9428499,12.0123349 L21.2694219,11.979737 C21.4486815,11.9581232 21.6719574,11.9039114 21.7642495,11.8550145 Z"
                              id="Shape"
                            />
                            <path
                              d="M26.6546653,7.32673772 L26.6546653,0.213658131 L28.3137931,0.213658131 L29.9675963,0.213658131 L30.0275862,0.430859516 C30.0549189,0.550267128 30.1092292,0.75116955 30.1418864,0.870577163 L30.2018763,1.09876263 L30.3594828,0.903175087 C30.56643,0.653375779 31.1755578,0.251570934 31.523428,0.132163322 C32.3774848,-0.155548789 33.4001521,0.0453536332 34.0639452,0.642746021 C34.3958418,0.941442215 34.5644523,1.22915433 34.6894016,1.7071391 C34.7820487,2.05473218 34.7873732,2.42925398 34.7873732,6.29529689 L34.7873732,10.5032637 L34.6404158,10.8182588 C34.3411765,11.453564 33.9223124,11.8011571 33.2368661,11.9747765 C32.7580122,12.0941841 31.9966024,12.0941841 31.5614097,11.9694616 C31.196856,11.8663529 30.7236815,11.5892706 30.4137931,11.3015585 C30.3158215,11.2037647 30.2235294,11.1279391 30.2125254,11.1279391 C30.2015213,11.1279391 30.1908722,11.8720221 30.1908722,12.7840554 L30.1908722,14.4401716 L28.4231237,14.4401716 L26.6553753,14.4401716 L26.6553753,7.32673772 L26.6546653,7.32673772 Z M30.9192698,9.96539792 C31.2454868,9.82969135 31.2454868,9.84599031 31.2674949,6.43064913 C31.278499,4.68772318 31.2621704,3.11310173 31.2348377,2.81972042 C31.1858519,2.36370381 31.1641988,2.29319308 31.0388945,2.18441522 C30.8539554,2.02709481 30.6147059,2.02709481 30.4350913,2.18441522 C30.2011663,2.38531765 30.1848377,2.72192664 30.2011663,6.32718616 C30.2174949,9.90551696 30.2174949,9.87291903 30.5110548,9.98134256 C30.7073529,10.0525619 30.7126775,10.0525619 30.9192698,9.96539792 Z"
                              id="Shape"
                            />
                          </g>
                        </g>
                      </g>
                    </g>
                  </svg>
                </div>
                <div
                  class="jsx-3066171714 labels__content"
                >
                  <span
                    class="jsx-4266605292 typography labels__typography"
                  >
                    6.1/10
                  </span>
                </div>
              </div>
              <div
                class="jsx-3066171714 labels labels__logo mb-3"
              >
                <div
                  class="jsx-4262843820 rotten-icon"
                >
                  <svg
                    height="1rem"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      id="Styleguide-presentation"
                      stroke="none"
                      stroke-width="1"
                    >
                      <g
                        fill="#FFFFFF"
                        fill-rule="nonzero"
                        id="icon__rotten"
                        transform="translate(-382.000000, -834.000000)"
                      >
                        <path
                          d="M387.35566,834.149397 C390.351365,833.370974 390.69458,835.791721 389.953222,837.773951 C389.20308,840.647422 392.028221,840.594028 392.238072,838.973753 C392.651443,836.77938 394.46203,835.792198 395.86975,837.02406 C397.314395,838.47033 396.680886,841.383845 393.825483,841.098524 C391.987283,840.914424 391.583406,843.432698 393.921687,843.62315 C397.617353,843.851335 398.794237,846.265051 397.481236,847.897602 C395.647164,849.703204 393.159,847.368566 392.262146,845.572963 C391.482384,844.463715 390.275903,845.586116 390.722897,846.347858 C392.21169,848.66773 391.058073,850.069783 389.472278,849.997321 C387.939221,849.896016 387.068742,848.917535 387.956996,846.523175 C388.403874,845.035549 386.971385,844.689326 386.056987,845.873634 C384.714401,848.210787 382.74685,847.50404 382.160759,846.373006 C381.788921,845.464603 381.8258,843.292516 384.926545,843.548757 C386.8402,844.041812 386.276929,842.051716 384.854416,841.923953 C383.794154,841.871155 381.356675,841.05345 382.232888,838.999353 C383.074699,837.681561 385.094884,837.596704 386.056987,839.749245 C386.405259,840.253693 387.015809,841.065249 388.125335,840.349158 C388.527835,839.729663 388.26418,838.692413 387.115196,837.949434 C385.554843,837.003203 385.702426,834.945173 387.35566,834.149397 Z"
                          id="path3575"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <div
                  class="jsx-3066171714 labels__content"
                >
                  <span
                    class="jsx-4266605292 typography labels__typography"
                  >
                    26%
                  </span>
                </div>
              </div>
              <button
                aria-label="Add to favourites"
                class="jsx-1783334891 button mb-3"
              >
                <svg
                  class="button__icon"
                  height="1.5rem"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="1.5rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Styleguide-presentation"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      id="8.icons"
                      transform="translate(-530.000000, -344.000000)"
                    >
                      <g
                        id="icon__heart"
                        transform="translate(530.000000, 344.000000)"
                      >
                        <rect
                          height="24"
                          id="Rectangle-13"
                          width="24"
                          x="0"
                          y="0"
                        />
                        <path
                          d="M12,5.7384007 C24.3159972,-3.87751835 26.3133219,14.4898505 12,20 C-2.31332192,15.570284 -0.31599724,-3.87751835 12,5.7384007 Z"
                          fill="currentColor"
                          id="Rectangle-12"
                          stroke="#FFFFFF"
                          stroke-width="2"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
                <span
                  class="jsx-4060198380 typography"
                >
                  Add to favourites
                </span>
              </button>
            </div>
            <div
              class="jsx-3039394109 mb-3"
            >
              <p
                class="jsx-1346695943 typography mb-2"
              >
                Plot
              </p>
              <p
                class="jsx-4060198380 typography"
              >
                Set in Sin City, story revolves around two people who discover they've gotten married following a night of debauchery, with one of them winning a huge jackpot after playing the other's quarter. Unhappy pair try to undermine each other and get their hands on the money -- falling in love along the way.
              </p>
            </div>
            <div
              class="jsx-3777709134 grid flex-direction-md justify-content-md"
            >
              <div
                class="jsx-2249794958 grid-item"
              >
                <div
                  class="jsx-1206893517 summary__credit mt-1"
                >
                  <p
                    class="jsx-1346695943 typography mb-1"
                  >
                    Cast
                  </p>
                  <ul
                    class="jsx-1206893517"
                  >
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Cameron Diaz
                      </p>
                    </li>
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Ashton Kutcher
                      </p>
                    </li>
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Rob Corddry
                      </p>
                    </li>
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Lake Bell
                      </p>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                class="jsx-2249794958 grid-item"
              >
                <div
                  class="jsx-1206893517 summary__credit mt-1"
                >
                  <p
                    class="jsx-1346695943 typography mb-1"
                  >
                    Genre
                  </p>
                  <ul
                    class="jsx-1206893517"
                  >
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Comedy
                      </p>
                    </li>
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Romance
                      </p>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                class="jsx-2249794958 grid-item"
              >
                <div
                  class="jsx-1206893517 summary__credit mt-1"
                >
                  <p
                    class="jsx-1346695943 typography mb-1"
                  >
                    Director
                  </p>
                  <ul
                    class="jsx-1206893517"
                  >
                    <li
                      class="jsx-1206893517"
                    >
                      <p
                        class="jsx-4060198380 typography"
                      >
                        Tom Vaughan
                      </p>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Movie detail page matches snapshot 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
